src/dom.js

```diff
@@ -536,6 +536,17 @@ export function ancestorElementsByTag(child, tagName) {
   });
 }
 
+/**
+ * Iterate over an array-like
+ * @param {!IArrayLike<T>} iterable
+ * @param {!function(T, number)} cb
+ * @template T
+ */
+export function iterateCursor(iterable, cb) {
+  for (let i = 0; i < iterable.length; i++) {
```

Comment:
In initial testing, using `i < iterable.length` [runs faster in jsperf](https://jsperf.com/iterating-over-collections-of-elements) than `iterable[i] != undefined`